README.md

```diff
@@ -1,2 +1,104 @@
 # action-beanstalk-deploy
 GitHub action for AWS Elastic Beanstalk deployment
+
+
+# Introduction
+This action manages a full beanstalk environment update. The whole steps are listed below:
+
+### Build and upload docker images
+This step builds docker images, given by a list of container configurations.
+The config file will be read from a config file, the path can be set with `config_path`
+An example of the configuration file is listed below.
+This step will only be executed if the `build_and_upload_image` key is set in the configuration file.
+
+### Create and upload deployment archive (Dockerrun.aws.json)
+Create `deploy-CommitHash.zip` and upload it to the specified application version bucket.
+This zip archive contains only the `Dockerrun.aws.json` at this moment.
+
+### Create Beanstalk application version
+This action uses a different Beanstalk application for each environment e.g. (dev, test, live)
+
+### Update Beanstalk environment
+Updates the given Beanstalk environment with given version label, which will be created before.
+
+### Wait until deployment finished
+Default settings will wait until the deployment process is finished and the Beanstalk environment is in ready and healthy state.
+
+
+# Config example
+`dev.json`:
+```json
+{
+  "build_and_upload_image": true,
+  "application_name": "my-sample-application-dev",
+  "environment_name": "sample-application-dev",
+  "application_version_bucket": "sample-application-version-bucket-dev",
+  "container_configs": [
```

Comment:
```suggestion
  "container": [
```
oder
```suggestion
  "container_settings": [
```

Reply:
Mit `containers` kann ich auch leben. In unserem Fall vom Mail Dispatcher ist es nur einer, für Keycloak haben wir an der stelle 2 Container, deshalb die Liste.

Wir verwenden tatsächlich ein und das selbe Image beim Deployment. Beim Deployment wird der `Commit Hash` als Tag an die `base_url` dran gehangen.